sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/json/JsonInferSchema.scala

```diff
@@ -68,21 +73,27 @@ private[sql] class JsonInferSchema(options: JSONOptions) extends Serializable {
             Some(inferField(parser))
           }
         } catch {
-          case  e @ (_: RuntimeException | _: JsonProcessingException) => parseMode match {
-            case PermissiveMode =>
-              Some(StructType(Seq(StructField(columnNameOfCorruptRecord, StringType))))
-            case DropMalformedMode =>
-              None
-            case FailFastMode =>
-              throw QueryExecutionErrors.malformedRecordsDetectedInSchemaInferenceError(e)
+          case e @ (_: JsonProcessingException | _: CharConversionException)
+              if ignoreCorruptFiles =>
+            logWarning("Skipped inferring json schema from the corrupted data: " +
+              s"${row.asInstanceOf[InternalRow].getString(0)}", e)
+            None
+          case e @ (_: RuntimeException | _: JsonProcessingException |
+                    _: CharConversionException) =>
```

Comment:
@yaooqinn can we just don't respect `ignoreCorruptFiles` here for now, and match here with `JacksonParser`:

https://github.com/apache/spark/blob/186815be1c0832e2f976faf6f2d5c31e42eebf1a/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/json/JacksonParser.scala#L495

https://github.com/apache/spark/blob/186815be1c0832e2f976faf6f2d5c31e42eebf1a/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/json/JacksonParser.scala#L490

meaning that here we will catch two more exceptions of `CharConversionException` (if encoding is not specified) and `MalformedInputException`.

I think JSON datasource itself doesn't really support `ignoreCorruptFiles` (not only JSON schema inference). For example, if other exceptions happen or `encoding` is specified, I think the exception will be thrown.

Reply:
thanks, can you take a look again to see if I did get your point